src/python/pants/backend/jvm/tasks/scalafix.py

```diff
@@ -31,6 +31,8 @@ def register_options(cls, register):
                   'providing the target classpath to scalafix. To enable this option, you '
                   'will need to install the `semanticdb-scalac` compiler plugin. See '
                   'https://www.pantsbuild.org/scalac_plugins.html for more information.')
+    register('--tool-classpath', type=str, default="", fingerprint=True,
```

Comment:
Do you check in the jar or have  common mount point you know will be on dev (and CI) machines? Typically we'd want to support resolving the jar and adding it to the classpath. In other words, we'd want the option to take a BUILD file address string and not a filesystem path. I think this would mean something like this here:
```python
cls.register_jvm_tool(register, 'scalafix-tool-classpath', classpath=[])
```

That says - register a classpath to resolve and give it a handle of 'scalafix-tool-classpath' allowing folks to customize it by default / convention with a root BUILD.tools file containing a `jar_library` target named 'scalafix-tool-classpath'.

And then access the resolved classpath like this down below using:
https://github.com/pantsbuild/pants/blob/039051735542d29ae02f4faa09c0c51c47292bf0/src/python/pants/backend/jvm/tasks/jvm_tool_task_mixin.py#L43-L54
```python
tool_classpath = self.tool_classpath('scalafix-tool-classpath')
if tool_classpath:
  args.append('--tool-classpath={}'.format(os.path.sep.join(tool_classpath)))
```